roxie/ccd/ccdqueue.cpp

```diff
@@ -610,38 +608,40 @@ void SlaveContextLogger::flush()
 
 //=================================================================================
 
-unsigned getIbytiDelay(unsigned channel, const RoxiePacketHeader &header)
+bool isSubChannelAlive(unsigned channel, unsigned subChannel)
 {
-    // MORE - adjust delay according to whether it's a retry, whether it was a broadcast etc
-    CriticalBlock b(ibytiCrit);
-    return IBYTIDelays[channel];
+    unsigned mask = SUBCHANNEL_MASK << (SUBCHANNEL_BITS * subChannel);
+    return (liveSubChannels[channel] & mask) == mask;
```

Comment:
Is this the correct test, or should it be checking for != 0, since I think the code 

    liveSubChannels[header.channel] |= header.retries & ROXIE_RETRIES_MASK

may not set all the bits if it is a retry

Reply:
It looks like the code that creates the packet header sets all the bits, and doesn't use the retries.

Reply:
The IBYTI message sets all bits - not entirely sure WHY, but it does.